os/pm/pm_procfs.c

```diff
@@ -146,6 +147,9 @@ static size_t power_metrics_read(FAR struct file *filep, FAR char *buffer, size_
 static size_t power_devices_read(FAR struct file *filep, FAR char *buffer, size_t buflen);
 static size_t power_lock_write(FAR struct file *filep, FAR const char *buffer, size_t buflen);
 static size_t power_unlock_write(FAR struct file *filep, FAR const char *buffer, size_t buflen);
+static size_t power_set_next_wakeup_interval(FAR struct file *filep, FAR const char *buffer, size_t buflen);
+static size_t power_get_last_wakeup_time(FAR struct file *filep, FAR char *buffer, size_t buflen);
+static size_t power_get_current_time(FAR struct file *filep, FAR char *buffer, size_t buflen);
```

Comment:
From these 2 functions, it seems to me that they are doing very little, and just returning the global pm timer parameters to the APP side.
Then the APP will use these returned values to set the next wakeup interval after the current wifi alive signal has been sent.

If my understanding here is correct, then I suggest avoiding these 2 functions which just return a parameter to the APP and have the main timer setting function power_set_next_wakeup_interval itself do the work.
The logic for identifying the wifi signal wakeup time based on elapsed time can be brought into the PM layer, where the parameters are available anyways.

Also, yet again, since this function is intended for setting the sleep interval for wifi alive signal wakeups, the name of the function may be changed to indicate that
(power_set_wifi_alive_wakup_interval OR power_set_signal_wakeup_interval maybe since we expect more such alive signals in the future to be handled).

Lastly, procfs needs to be substituted by a driver dedicated to PM, so these comments may be considered there instead.

Reply:
I should add these when driver dedicated to PM is created.